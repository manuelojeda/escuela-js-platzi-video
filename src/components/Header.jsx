import React from 'react';
import '../assets/styles/components/Header.scss';
import Logo from '../assets/images/logo-platzi-video-BW2.png';
import ProfilePic from '../assets/images/user-icon.png';

const Header = () => (
  <header className='header'>
    <img className='header__img' src={Logo} alt='Platzi Video' />
    <div className='header__menu'>
      <div className='header__menu--profile'>
        <img src={ProfilePic} alt='' />
        <p>Perfil</p>
      </div>
      <ul>
        <li><a href='/'>Cuenta</a></li>
        <li><a href='/'>Cerrar SesiÃ³n</a></li>
      </ul>
    </div>
  </header>
);

export default Header;
