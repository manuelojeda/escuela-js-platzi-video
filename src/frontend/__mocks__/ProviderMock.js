import React from 'react';
import { Provider } from 'react-redux';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import initialState from '../initialState'
import { createBrowserHistory } from 'history';
import reducer from '../reducers';

const store = createStore(reducer, initialState, compose(applyMiddleware(thunk)))
const history = createBrowserHistory();

const ProviderMock = (props) => (
  <Provider store={store}>
      <Router history={history}>
        {props.children}
      </Router>
  </Provider>
)

export default ProviderMock
