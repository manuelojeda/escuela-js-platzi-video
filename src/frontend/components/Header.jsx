/* eslint-disable react/jsx-wrap-multilines */
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { logoutRequest } from '../actions';
import '../assets/styles/components/Header.scss';
import Logo from '../assets/images/logo-platzi-video-BW2.png';
import ProfilePic from '../assets/images/user-icon.png';
import gravatar from '../utils/gravatar';

const Header = (props) => {
  const { user } = props;

  const hasUser = Object.keys(user).length > 0;

  const handleLogout = () => {
    props.logoutRequest({});
  };

  return (
    <header className='header'>
      <Link to='/'>
        <img className='header__img' src={Logo} alt='Platzi Video' />
      </Link>
      <div className='header__menu'>
        <div className='header__menu--profile'>
          {hasUser ?
            <img src={gravatar(user.email)} alt={user.email} /> :
            <img src={ProfilePic} alt='User Icon' />}
          <p>{user.email}</p>
        </div>
        <ul>
          {hasUser ?
            <li><a href='/'>Cuenta</a></li> :
            null}

          {hasUser ?
            <li><a href='#logout' onClick={handleLogout}>Cerrar Sesión</a></li> :
            <li>
              <Link to='/login'>
                Iniciar Sesión
              </Link>
            </li>}

        </ul>
      </div>
    </header>
  );
};

const mapStateToProps = (state) => {
  return {
    user: state.user,
  };
};

const mapDispatchToProps = {
  logoutRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
