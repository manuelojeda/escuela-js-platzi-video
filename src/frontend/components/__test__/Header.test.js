import React from 'react'
import { create } from 'react-test-renderer'
import ProviderMock from '../../__mocks__/ProviderMock'
import Header from '../Header'

describe('Header testing', () => {
    test('Match snapshot', () => {
      const header = create(
        <ProviderMock>
          <Header />
        </ProviderMock>
      )

      expect(header.toJSON()).toMatchSnapshot();
    })
})
