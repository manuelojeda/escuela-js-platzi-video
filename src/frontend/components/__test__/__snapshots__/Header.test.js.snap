// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header testing Match snapshot 1`] = `
<header
  className="header"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Platzi Video"
      className="header__img"
      src="test-file-stubs"
    />
  </a>
  <div
    className="header__menu"
  >
    <div
      className="header__menu--profile"
    >
      <img
        alt="User Icon"
        src="test-file-stubs"
      />
      <p />
    </div>
    <ul>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Iniciar Sesi√≥n
        </a>
      </li>
    </ul>
  </div>
</header>
`;
